lib/gui/app.js

```diff
@@ -100,9 +101,12 @@ app.run((AnalyticsService, ErrorService, UpdateNotifierService, SelectionStateMo
     currentVersion: packageJSON.version
   });
 
-  UpdateNotifierService.isLatestVersion().then((isLatestVersion) => {
+  const CURRENT_VERSION = ManifestBindService.get('version');
 
-    if (isLatestVersion) {
+  s3Packages.getLatestVersion(s3Packages.BUCKET_URL_PRODUCTION, {
+    currentVersion: CURRENT_VERSION
+  }).then((latestVersion) => {
+    if (latestVersion === CURRENT_VERSION) {
```

Comment:
This is looking much better :+1: , but IMHO it doesn't make sense to pass in the `currentVersion` as a parameter to `getLatestVersion()`. I think it'd be much clearer if you just did the semver-comparison inside this function, rather than hiding it away inside `s3Packages`. It makes s3Packages a bit simpler, and it allows you to do much "richer" logic here, should the need arise in future. (and for clarity, we should rename `latestVersion` to `latestS3Version`, again IMHO)